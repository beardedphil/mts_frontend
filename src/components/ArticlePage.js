import React from 'react';

const ArticlePage = () => {
    return (
        <div className="article-page">
            <h1>ArticlePage Component</h1>
        </div>
    );
};

export default ArticlePage;
