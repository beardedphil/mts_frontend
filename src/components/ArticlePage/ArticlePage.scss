.field {
  display: flex;
  justify-content: center;
  margin-bottom: 15px;
}

.field input {
  width: 250px;
}

@media (min-width: 768px) {
  .field input {
    width: 350px;
  }
}

