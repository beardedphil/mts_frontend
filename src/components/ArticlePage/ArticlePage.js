import React, { useState, useEffect } from 'react';
import axios from 'axios';
import ArticleList from '../ArticleList/ArticleList';
import { useTrustedSources } from '../contexts/TrustedSourceContext';
import './ArticlePage.scss';

const ArticlePage = () => {
    const [terms, setTerms] = useState('');
    const [articles, setArticles] = useState();
    const [articlesLink, setArticlesLink] = useState();
    const [keywords, setKeywords] = useState([]);
    const [trustedSources] = useTrustedSources();

    const getMoreArticles = async () => {
        const sources = trustedSources.length > 0 ? trustedSources.toString() : null;
        return await axios.get(articlesLink, {
            params: {
                keywords,
                sources
            }
        });
    };

    const scrollLoadArticles = () => {
        getMoreArticles().then((response) => {
            if (response) {
                setArticles([...articles, ...response.data.results]);
                setArticlesLink(response.data.next);
            }
        }).catch((err) => {
            console.log(err);
        });
    }

    useEffect(() => {
        const timerId = setTimeout(() => {
            setKeywords(terms);
        }, 500);

        return () => {
            clearTimeout(timerId);
        }
    }, [terms]);

    useEffect(() => {
        const search = async () => {
            const sources = trustedSources.length > 0 ? trustedSources.toString() : null;
            return await axios.get('https://mytrustedsourceapi.herokuapp.com/articles/', {
                params: {
                    keywords,
                    sources
                }
            });
        };
        search().then((response) => {
            if (response) {
                setArticles(response.data.results);
                setArticlesLink(response.data.next);
            }
        }).catch((err) => {
            console.log(err);
        });
    }, [keywords, trustedSources]);


    return (
        <div className="article-page">
            <div className="form">
                <div className="field">
                    <input
                        data-testid="search"
                        id="search"
                        onChange={e => setTerms(e.target.value)}
                        placeholder="Search..."
                        type="text"
                        value={terms}
                    />
                </div>
            </div>

            <ArticleList articles={ articles } next={ scrollLoadArticles } hasMore={ !!articlesLink } />
        </div>
    );
};

export default ArticlePage;
