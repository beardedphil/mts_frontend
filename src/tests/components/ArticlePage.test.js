import React from 'react';
import { shallow } from 'enzyme';
import ArticlePage from '../../components/ArticlePage';

test('should render ArticlePage correctly', () => {
    const wrapper = shallow(<ArticlePage />);
    expect(wrapper).toMatchSnapshot();
});
