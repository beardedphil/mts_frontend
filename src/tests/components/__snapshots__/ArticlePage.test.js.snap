// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ArticlePage correctly 1`] = `
<div
  className="article-page"
>
  <h1>
    ArticlePage Component
  </h1>
</div>
`;
